jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -1603,6 +1604,118 @@ ecma_builtin_typedarray_prototype_find_index (ecma_value_t this_arg, /**< this a
   return ecma_builtin_typedarray_prototype_find_helper (this_arg, predicate, predicate_this_arg, false);
 } /* ecma_builtin_typedarray_prototype_find_index */
 
+/**
+ * The %TypedArray%.prototype object's 'indexOf' and 'lastIndexOf' routine helper
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_index_helper (ecma_value_t this_arg, /**< this argument */
+                                                const ecma_value_t args[], /**< arguments list */
+                                                ecma_length_t args_number, /**< number of arguments */
+                                                bool last_index_of) /**< true - lastIndexOf routine
+                                                                         false - indexOf routine */
+{
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+
+  ecma_object_t *typedarray_p = ecma_get_object_from_value (this_arg);
+  uint32_t length = ecma_typedarray_get_length (typedarray_p);
+  lit_magic_string_id_t class_id = ecma_object_get_class_name (typedarray_p);
+  lit_utf8_byte_t *typedarray_buffer_p = ecma_typedarray_get_buffer (typedarray_p);
+  uint8_t shift = ecma_typedarray_get_element_size_shift (typedarray_p);
+  uint8_t element_size = (uint8_t) (1 << shift);
+  uint32_t limit = length * element_size;
+
+  if (length == 0)
+  {
+    return ecma_make_integer_value (-1);
+  }
+
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  uint32_t from_index;
+
+  if (args_number == 1)
+  {
+    from_index = last_index_of ? length - 1 : 0;
+  }
+  else
+  {
+    ecma_number_t num_var;
+    ret_value = ecma_get_number (args[1], &num_var);
```

Comment:
Based the standard this should be the `toInteger` conversion? If so, do we have tests for `Nan`, `+/-infinity`, string values, objects, etc as arguments?

Reply:
Please check the result of the to_number operation, since it can throw error.